src/frontend/containers/app/page/blocks/youtubeDownload.tsx

```diff
@@ -100,7 +99,17 @@ export const Component = (props: { db: Types.Database.LoadedDatabase; block: Typ
 export const isAvailable = (
   db: Types.Database.LoadedDatabase,
   config: Types.Config.PartialConfig,
-): Observable<boolean> => DownloadService.isAvailable(db.instance, config);
+): Observable<boolean> => {
+  const dataAvailable = db.instance.docs
+     .find({
+       selector: {
+         data__type: Types.Data.Type.youtube_url,
+       },
+     }).$.pipe(map(datas => datas.map(data => data.get())))
+  .pipe(mergeMap(datas => DownloadService.isAvailable(datas, config)))
```

Comment:
additionally, the `Service.isAvailable` no longer gets the db, and just gets data instead.

the db querying gets moved to `Block.isAvailable`

Reply:
This is going to be useful because we can test service availability on a _subset_ of data.  i.e. "is this service available given the data _in this block_?"

Reply:
instead of always testing availability on the entire database